realm/realm-library/src/androidTest/java/io/realm/internal/OsSharedRealmTests.java

```diff
@@ -178,6 +178,33 @@ public void onSchemaChanged() {
         assertTrue(listenerCalled.get());
     }
 
+    // Test for https://github.com/realm/realm-core/issues/3707
+    @Test
+    public void emitTableInstructionsForCustomClasses() {
+        final AtomicBoolean listenerCalled = new AtomicBoolean(false);
+        assertFalse(sharedRealm.hasTable("NewTable"));
+        sharedRealm.registerSchemaChangedCallback(new OsSharedRealm.SchemaChangedCallback() {
+            @Override
+            public void onSchemaChanged() {
+                assertTrue(sharedRealm.hasTable("NewTable"));
+                listenerCalled.set(true);
+            }
+        });
+
+        // Change schema using another Realm
+        // Classes not starting with class_ were treated differently by Sync
+        OsSharedRealm bgRealm = OsSharedRealm.getInstance(config, OsSharedRealm.VersionID.LIVE);
+        bgRealm.beginTransaction();
+        bgRealm.createTable("NewTable");
+        bgRealm.commitTransaction();
```

Comment:
I assume the callback above will be immediately executed after committing the transaction, right? Otherwise, we could face problems with the test finishing before the callback kicks in.

Reply:
Yes, it does.